core/variant_op.cpp

```diff
@@ -521,7 +521,7 @@ void Variant::evaluate(const Operator &p_op, const Variant &p_a,
 				const Dictionary *arr_a = reinterpret_cast<const Dictionary *>(p_a._data._mem);
 				const Dictionary *arr_b = reinterpret_cast<const Dictionary *>(p_b._data._mem);
 
-				_RETURN((*arr_a == *arr_b) == false);
```

Comment:
I just want to say, I have no idea what this person was thinking. "If equal, then if that equality is equal to false return true"? Much clearly written as "If not equal return true".

Reply:
Well given that the `!=` didn't exist, it's understandable, though it could have been written as `!(*arr_a == *arr_b)`.